add walmart icon

sachinraja opened the pull request:
<!--
Before opening your pull request, have a quick look at our contribution guidelines:
https://github.com/simple-icons/simple-icons/blob/develop/CONTRIBUTING.md

Consider adding a preview image of your submission using:
https://petershaggynoble.github.io/SI-Sandbox/preview/
-->
![walmart](https://user-images.githubusercontent.com/58836760/129135256-b316dda6-a48a-4fb8-b8a3-e39aa9c2017d.png)
**Issue:** closes #6253
**Alexa rank:** [95](https://www.alexa.com/siteinfo/walmart.com)
  <!-- The Alexa rank can be retrieved at https://www.alexa.com/siteinfo/
       Please see our contributing guidelines for more details on how we
       assess a brand's popularity. -->

### Checklist
  - [x] I updated the JSON data in `_data/simple-icons.json`
  - [x] I optimized the icon with SVGO or SVGOMG
  - [x] The SVG `viewbox` is `0 0 24 24`

### Description
<!--
Anything relevant, for example:
  - Why did you pick the hex value?
  - Did you manually vectorize the logo?
  - Have you used multiple sources?
  - etc.
-->
I chose `#0071CE` as the hex value because I found the color [here](https://one.walmart.com/content/dam/px/associate_brand_center/all-company-brand-guidelines/ab_color_palette_191016.pdf). They even call it "Walmart Blue." It also shows up a few places throughout their site (see https://walmart.com below the navbar for a similar color and https://corporate.walmart.com/ navbar for the exact color). This logo includes the registered trademark symbol as I found it in the logos used on their site.

sachinraja commented:
Not exactly sure how to fix this because when I remove the collinear segment `m6.532 0`, I get two more errors:
```
icon-size Size of <path> must be exactly 24 in one dimension; the size is currently 26.446 x 5.656
icon-centered <path> must be centered at (12, 12); the center is currently (10.777, 12)
 ```

jorgeamadosoria commented:
> Not exactly sure how to fix this because when I remove the collinear segment `m6.532 0`, I get two more errors:
> 
> ```
> icon-size Size of <path> must be exactly 24 in one dimension; the size is currently 26.446 x 5.656
> icon-centered <path> must be centered at (12, 12); the center is currently (10.777, 12)
> ```

that's fixable on inkscape, I've run into that a couple times.
However when I remove the collinear that's not what I, I just get the icon scrambled. Can you provide the svg code after removing the segment so I can adjust it on inkscape?

sachinraja commented:
I also get the icon scrambled

jorgeamadosoria commented:
> I also get the icon scrambled

ok, we are on the same page then

sachinraja commented:
Updated icon according to https://github.com/simple-icons/simple-icons/issues/6272#issuecomment-897600708
![walmart](https://user-images.githubusercontent.com/58836760/129212330-90fa7235-c170-4f4c-afc2-42c054b348b0.png)

@jorgeamadosoria  The problem was the attributes on the XML document (like `stroke-width`). I had to go to the XML Editor and remove them.

jorgeamadosoria commented:
> Updated icon according to [#6272 (comment)](https://github.com/simple-icons/simple-icons/issues/6272#issuecomment-897600708)
> ![walmart](https://user-images.githubusercontent.com/58836760/129212330-90fa7235-c170-4f4c-afc2-42c054b348b0.png)
> 
> @jorgeamadosoria The problem was the attributes on the XML document (like `stroke-width`). I had to go to the XML Editor and remove them.

oh. I had just solved it by moving around the letters:
`<svg role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title>Walmart</title><path d="M23.747 13.517h.027c.03 0 .057-.01.057-.036 0-.021-.013-.036-.053-.036l-.031.003zm0 .124h-.041v-.221a.421.421 0 0 1 .076-.007c.038 0 .056.007.07.017.014.01.023.027.023.047 0 .028-.02.045-.044.053v.003c.02.007.03.025.037.053.002.02.008.038.016.055h-.042a.152.152 0 0 1-.017-.058c-.005-.024-.02-.035-.052-.035h-.026zm.037-.29a.17.17 0 0 0-.169.174.17.17 0 0 0 .171.174c.096.002.166-.076.166-.173 0-.097-.07-.175-.167-.175zm.002-.037a.21.21 0 0 1 .214.21.211.211 0 0 1-.215.214c-.283 0-.283-.424 0-.424h.001m-3.686-.852c.067.115.062.244-.008.301l-1.216.853c-.127.073-.316-.016-.424-.202-.108-.186-.089-.395.037-.468l1.347-.626c.084-.032.199.028.265.142m1.6 0c.066-.114.18-.174.264-.142l1.347.626c.127.073.145.282.038.468-.108.186-.298.275-.424.202l-1.216-.853c-.07-.057-.075-.186-.009-.3m-.8.462c.132 0 .24.068.256.157l.13 1.48c0 .146-.17.266-.386.266s-.387-.12-.387-.266l.131-1.48c.015-.089.124-.157.256-.157m.8-1.387c.066.114.18.174.264.143l1.347-.627c.127-.073.145-.281.038-.468-.108-.186-.298-.274-.424-.201l-1.216.853c-.07.057-.075.186-.009.3m-1.6 0c.066-.114.06-.243-.01-.3l-1.215-.853c-.127-.073-.316.015-.424.201-.108.187-.089.395.037.468l1.347.627c.084.031.199-.029.265-.143m.8-.462c.132 0 .24-.068.256-.158l.13-1.48c0-.145-.17-.266-.386-.266s-.387.12-.387.267l.131 1.48c.015.089.124.157.256.157m-13.094.1v2.465h.67v-1.446c0-.067.007-.14.03-.201.056-.146.192-.316.408-.316.27 0 .397.228.397.558v1.405h.67v-1.463c0-.065.008-.143.027-.2a.423.423 0 0 1 .402-.3c.274 0 .406.224.406.612v1.351h.67V12.19c0-.766-.39-1.069-.828-1.069a.893.893 0 0 0-.487.134 1.038 1.038 0 0 0-.312.307h-.01a.703.703 0 0 0-.68-.44c-.417 0-.604.21-.718.39h-.01v-.335m6.035 .02v2.465h.688a32.817 32.817 0 0 1-.005-1.44c.175-.495.437-.426.762-.42v-.646a.723.723 0 0 0-.822.505v-.464h-.603m1.72-.817v2.422c0 .333.064.567.198.71.118.125.31.206.543.206.197 0 .39-.037.482-.072l-.009-.52c-.068.016-.146.03-.253.03-.227 0-.303-.146-.303-.445v-.927h.58v-.628h-.58v-.776h-.657m-8.72 2.585V10.36h-.667v3.282h.667v-.697m-1.777-.183c0 .045-.005.09-.017.133a.467.467 0 0 1-.45.315c-.184 0-.33-.104-.33-.326 0-.339.373-.432.797-.43zm.666-.59c0-.56-.239-1.05-1.045-1.05a1.91 1.91 0 0 0-.922.219l.13.45c.165-.104.426-.19.673-.19.41-.001.477.232.477.38v.036c-.893-.001-1.456.307-1.456.937 0 .384.287.744.786.744.307 0 .563-.123.717-.319h.015s.102.426.664.263a3.603 3.603 0 0 1-.04-.594v-.877M0 10.36s.583 2.387.676 2.775c.108.452.304.619.868.506l.364-1.48c.092-.368.154-.63.213-1.004h.01c.042.378.102.637.178 1.005 0 0 .148.672.223 1.025.077.353.288.575.84.454l.868-3.282h-.7l-.296 1.42c-.08.412-.152.735-.208 1.113h-.01c-.05-.374-.115-.684-.196-1.087l-.308-1.446h-.73l-.329 1.41c-.093.428-.18.774-.236 1.14h-.01c-.066-.4-.137-.797-.215-1.194L.737 10.36H0m13.23 2.401c0 .045-.005.09-.017.134a.467.467 0 0 1-.45.315c-.184 0-.331-.105-.331-.326 0-.339.373-.432.797-.43zm.665-.59c0-.558-.238-1.05-1.045-1.05a1.91 1.91 0 0 0-.922.22l.131.449a1.35 1.35 0 0 1 .673-.189c.409-.001.476.232.476.38v.036c-.892-.001-1.456.307-1.456.937 0 .384.287.744.786.744.307 0 .564-.123.718-.319h.015s.102.426.663.263c-.061-.496-.039-.953-.039-1.471"/></svg>`

But great! let's go! :)

jorgeamadosoria commented:
> Updated icon according to [#6272 (comment)](https://github.com/simple-icons/simple-icons/issues/6272#issuecomment-897600708)
> ![walmart](https://user-images.githubusercontent.com/58836760/129212330-90fa7235-c170-4f4c-afc2-42c054b348b0.png)
> 
> @jorgeamadosoria The problem was the attributes on the XML document (like `stroke-width`). I had to go to the XML Editor and remove them.

@sachinraja Reading #6272 , I think the hex @service-paradis suggested is not correct. The Spark brand guidelines seem to point strongly towards the yellow rather than the dark blue. I would recommend a second reading there to see what you think?
I also made this comment in response to @service-paradis comment in the issue.

sachinraja commented:
Yeah, the spark is definitely yellow. I'll wait for a confirmation though

service-paradis commented:
As seen in our [guidelines](https://github.com/simple-icons/simple-icons/blob/develop/CONTRIBUTING.md#color-guidelines), the color should be the brand color, it doesn't have to be in the logo itself.

service-paradis commented:
For the icon itself, I'm not so sure anymore if we should use the spark or the full wordmark 😅 

The following sentence from their [guidelines](https://one.walmart.com/content/people-experience/associate-brand-center.html.html) makes me rethink this one. We might get the help of other @simple-icons/maintainers on this one! Sorry @sachinraja! 

> The full Walmart logo represents our brand. We're proud to call ourselves Walmart, and the full logo should be used when materials are outside of the Walmart context.

runxel commented:
Yeah, this really looks like we should include the wordmark, too.

> 
> For the icon itself, I'm not so sure anymore if we should use the spark or the full wordmark 😅
> 
> The following sentence from their [guidelines](https://one.walmart.com/content/people-experience/associate-brand-center.html.html) makes me rethink this one. We might get the help of other @simple-icons/maintainers on this one! Sorry @sachinraja!
> 
> > The full Walmart logo represents our brand. We're proud to call ourselves Walmart, and the full logo should be used when materials are outside of the Walmart context.

sachinraja commented:
@service-paradis I think this is a bit of a gray area because those aren't actually their brand guidelines, but an internal Walmart resource.

jorgeamadosoria commented:
> As seen in our [guidelines](https://github.com/simple-icons/simple-icons/blob/develop/CONTRIBUTING.md#color-guidelines), the color should be the brand color, it doesn't have to be in the logo itself.

yellow is a brand color,though. It's one of five in the palette. And specifically it's the one recommended in their own page for use when using the spark.
A blue spark does not represent Walmart, neither popularly nor going by their own brand guidelines. Blue Walmart text, yes, Blue Walmart + spark, compromise, Blue spark... not so sure.

jorgeamadosoria commented:
> Yeah, this really looks like we should include the wordmark, too.
> 
> > For the icon itself, I'm not so sure anymore if we should use the spark or the full wordmark 😅
> > The following sentence from their [guidelines](https://one.walmart.com/content/people-experience/associate-brand-center.html.html) makes me rethink this one. We might get the help of other @simple-icons/maintainers on this one! Sorry @sachinraja!
> > > The full Walmart logo represents our brand. We're proud to call ourselves Walmart, and the full logo should be used when materials are outside of the Walmart context.

I think this runs into the whole "logo vs icon" discussion.
Since they seem to provide the spark as an identificative item, maybe it's fine if we use just that? We don't aim to provide their logo, rather, just an icon to identify the brand, right?
Maybe a maintainer can weigh in.

jorgeamadosoria commented:
> For the icon itself, I'm not so sure anymore if we should use the spark or the full wordmark 😅
> 
> The following sentence from their [guidelines](https://one.walmart.com/content/people-experience/associate-brand-center.html.html) makes me rethink this one. We might get the help of other @simple-icons/maintainers on this one! Sorry @sachinraja!
> 
> > The full Walmart logo represents our brand. We're proud to call ourselves Walmart, and the full logo should be used when materials are outside of the Walmart context.

If that' the case, then I've provided a SVG text for @sachinraja to validate if they choose to (it deals with the collinear problem they were having) and in that case then the hex is applicable, I think.

service-paradis commented:
The hex we provide only serves as a 'guide' to end-users anyway, as the SVG is served without color (defaulting to the path being black), with the hex being available via the JSON. We aim to provide the primary brand color, whatever if it makes sense with the icon or not. In this case, yellow is not Walmart primary color whatever icon we choose.

Like you mention, we can wait for another @simple-icons/maintainers  to weight in about the logo and the color.

ericcornelissen commented:
I must agree that I would have expected yellow as the brand color for Walmart, but [the color palette](https://one.walmart.com/content/dam/px/associate_brand_center/all-company-brand-guidelines/ab_color_palette_191016.pdf) shared by @sachinraja in the Pull Request description does strongly suggest that blue is the "correct" choice. To add to that, the spark in the bottom right of that file is in many different colors, further suggesting it doesn't _have_ to be yellow. Also considering that the brand color `!=` the color of the SVG I would agree that blue is indeed the correct choice here.

jorgeamadosoria commented:
> I must agree that I would have expected yellow as the brand color for Walmart, but [the color palette](https://one.walmart.com/content/dam/px/associate_brand_center/all-company-brand-guidelines/ab_color_palette_191016.pdf) shared by @sachinraja in the Pull Request description does strongly suggest that blue is the "correct" choice. To add to that, the spark in the bottom right of that file is in many different colors, further suggesting it doesn't _have_ to be yellow. Also considering that the brand color `!=` the color of the SVG I would agree that blue is indeed the correct choice here.

I remain unconvinced. The yellow is called Spark Yellow. It's kind of it's only reason of being there.
Also, the palette says _"Color hierarchy, context of message, and media channel need to be considered when producing a piece of communication for the Associate Brand"_
The idea is not that the colors are ranked in importance, but rather that when using this colors in context, they need to be considered in order.
An example of what I mean: 
https://one.walmart.com/content/dam/px/associate_brand_center/all-company-brand-guidelines/AssociateBrand_VISID_200203.pdf

You would notice the use of the blue ink as a background color for Sections (to bring up the white), of Walmart blue for headers (as expected) and of the little yellow spark in each and every page.
It is my very firm opinion that the Walmart spark cannot be represented by any color other than the designated Spark Yellow. It is a color solely designed for it, it is used as such in Walmart's own brand related documentation, and I could not remember having ever seen a Walmart spark of any other color, except the one in the Palette @sachinraja linked, which I don't count because it is there to showcase all colors. 

Having said that, I have zero power of decision on this. As @service-paradis said, the hex is a suggestion, I can always use any other color when using the library, so it's not a deal breaker.
I just think `#ffc220` is better than `#041f41` if the icon is going to be the spark only.

I think I've have exhausted my arguments. I'll accept whatever the ultimately result turns out to be.

sachinraja commented:
@ericcornelissen [as outlined by @service-paradis](https://github.com/simple-icons/simple-icons/pull/6269#issuecomment-897856008) do you think we should use the wordmark or just the spark here?

ericcornelissen commented:
> do you think we should use the wordmark or just the spark here?

Based on the "Logo creation" tab of the [Associate Brand Center](https://one.walmart.com/content/people-experience/associate-brand-center.html.html) (linked by @service-paradis) it seems to me that we should include the wordmark. By and large people that get the logo through simple icons will be using it "Out of context" :thinking: 


> About the hex value, I'm a bit confused on why the brand color is suggested and not the actual logo color, though that may be a question better asked in a separate issue.

Primarily because that is what we have done for the other 1900+ icons. On top of that, as @service-paradis, already noted, the color is not embedded in the SVG, and not every user will use the brand color as the color for the SVG. For example, one could choose to embed the SVG in a button that has the brand color as background color. If you want to discuss in more depth I would indeed encourage you to open an issue, after looking for existing issues on the topic as this isn't the first time it has come up.

sachinraja commented:
Seems like three maintainers are in agreement here so I will revert to the wordmark. I still think "Walmart Blue" (see [color palette](https://one.walmart.com/content/people-experience/associate-brand-center.html.html)) would be a better color to represent the brand. This may sound weird, but this is the color that I (and probably most people) think of when they think about Walmart. It is also the one used the most in their stores (on signs, on the storefronts, etc).

jorgeamadosoria commented:
> Seems like three maintainers are in agreement here so I will revert to the wordmark. I still think "Walmart Blue" (see [color palette](https://one.walmart.com/content/people-experience/associate-brand-center.html.html)) would be a better color to represent the brand. This may sound weird, but this is the color that I (and probably most people) think of when they think about Walmart. It is also the one used the most in their stores (on signs, on the storefronts, etc).

if we are going with the wordmark in full, then I also think Walmart blue is in order rather thank Blue Ink or Spark Yellow.

ericcornelissen commented:
Just to clarify, I agree blue is the best choice

sachinraja commented:
@jorgeamadosoria You don't have to keep this updated with the main branch. We only need to update if there is a merge conflict once the PR is ready to committed

jorgeamadosoria commented:
> @jorgeamadosoria You don't have to keep this updated with the main branch. We only need to update if there is a merge conflict once the PR is ready to committed

I know, it's a mania :)
I think we are all in agreement, should we merge this already?

sachinraja commented:
![walmart (1)](https://user-images.githubusercontent.com/58836760/129513197-b5bf7b8f-16a2-4c14-8a26-8f8f1c585d57.png)
Well this is mergeable now that I've updated htis. I've changed the color to `#0071CE` as I feel this fits the Walmart brand more. Thanks to @jorgeamadosoria for providing the fixed path!

ericcornelissen commented on icons/walmart.svg:
```diff
@@ -0,0 +1 @@
+<svg role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title>Walmart</title><path d="M20.598 12.462c.068.115.063.244-.008.301l-1.247.853c-.13.073-.323-.016-.434-.202-.11-.186-.091-.395.038-.468l1.38-.626c.086-.032.204.028.272.142m1.64 0c.067-.114.184-.174.27-.142l1.38.626c.13.073.149.282.04.468-.111.186-.306.275-.435.202l-1.246-.853c-.072-.057-.077-.186-.01-.3m-.82.462c.136 0 .24 ...
```
This is what I get when I compare to [the current source]() after removing the `(R)` :thinking: 

![image](https://user-images.githubusercontent.com/3742559/129858420-cd9b592f-cc73-4102-ac77-b9271a1f6015.png)

sachinraja replied:
This is probably due to the readjustment we did after removing the collinear segment. I'll try to realign it.

sachinraja replied:
@jorgeamadosoria I'd appreciate any help on this since you seem to know how to fix these issues. I'm honestly unsure how to realign this.

jorgeamadosoria replied:
let me try, hold on

jorgeamadosoria replied:
I remember this movement. It's because of the collinear elimination, 4 letters move. apparently some other things happen too

jorgeamadosoria replied:
![image](https://user-images.githubusercontent.com/15719479/130004959-2f6d4358-e00b-4eca-8889-759863238fab.png)

This is the difference I get.
I removed the registered trademark, Path -> union the paths, optimized with svgo (precision 5) and then overimpose the source with fill: `#ffffff`

That's the result. No collinear segments or anything to remove now, but the outline seems to suggest that the svgo version (black) is slightly larger than the source prior to transformation (white).
I'm perfectly fine accepting this as is, but probably other maintainer are more stringent.

ericcornelissen replied:
@jorgeamadosoria, is that comparison of the SVG in this Pull Request and the you get after "I removed the registered trademark, Path -> union the paths, optimized with svgo (precision 5)"? Or is that the source SVG compared against the latter? 

Or, perhaps a simpler question, do you see the same differences as I do when comparing the source SVG to the SVG in the Pull Request (as suggested by [this comment](https://github.com/simple-icons/simple-icons/pull/6269#discussion_r691752050))?

jorgeamadosoria replied:
Firat option. The source needs to be modificed to be compared

ericcornelissen replied:
What should the source be then?

jorgeamadosoria replied:
@ericcornelissen I didn't mean the source is wrong or anything, I just meant that some changes needed to be applied before comparison.

So, I did something on a hunch: I took the same path twice, fill it once with `#fff` and once with `#000`, and overimpose them.
the result is as follows:
![image](https://user-images.githubusercontent.com/15719479/130340753-f603008e-a2d5-4d69-8697-33c0e5f40539.png)

This is the svg code:
`<svg role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title>Walmart</title>
  <path id="propos" fill="#000" d="M21.418184 9.1021869 c -.220482 0 -.395839 .1230808 -.395839 .2729758 l .133933 1.5162683 c.01478 .09132 .126688 .161852 .261967 .161852 .135555 -1.72e-4 .247054 -.07065 .262149 -.161852 l .134238 -1.5162683 c 0 -.149895 -.17555 -.2729758 -.396448 -.2729758 z M -1.4486275e-5 10.318413 c 0 0 .597125316275 2.446983 .692414506275 2.844167  .11123373 .463624 .31170438 .63419 .88954338 .519128 l .3729088 -1.51718 c .094496 -.376831 .157901 -.64553 .2186604 -1.02883 h .010644 c .042693 .387111 .1035396 .653142 .1813148 1.030168 0 0 .1517616 .688693 .229487 1.050421 .07795 .361633 .2948212 .589502 .8608346 .465421 L 4.3443025 10.318413 H 3.6269506 L 3.323563 11.77252 c -.081554 .423252 -.1554409 .75396 -.2125173 1.14117 H 3.1008274 C 3.0489413 12.530225 2.9830587 12.21273 2.9001099 11.800377 L 2.5842536 10.318413 H 1.8368546 L 1.499041 11.763032 c -.09568 .438984 -.1852681 .793372 -.2421991 1.167445 h -.010218 c -.058322 -.35224 -.1359895 -.797691 -.2200594 -1.222612 0 0 -.20074192 -1.033275 -.27115107 -1.389452 z m 6.838444886275 0 v 3.363295 h .6829868 v -.714067 -2.649228 z m 9.6188076 0 v 2.481173 c 0 .342022 .06439 .581706 .202117 .728117 .120331 .128055 .318542 .210936 .556048 .210936 .201923 0 .400619 -.0383 .494252 -.07317 l -.0088 -.533665 c -.06968 .01711 -.149794 .03078 -.259413 .03078 -.232648 0 -.310686 -.148997 -.310686 -.456115 v -.949212 h .594793 v -.643512 h -.594812 v -.795328 z m 2.778843 0 c -.114465 .0027 -.244518 .08936 -.327231 .232772 -.110615 .19096 -.09105 .404339 .03838 .479228 l 1.379902 .642539 c .0862 .03205 .203231 -.02912 .27103 -.145976 .06814 -.117446 .06289 -.249593 -.0088 -.308193 l -1.246353 -.874094 c -.03237 -.01875 -.06877 -.02718 -.106928 -.02628 z m 4.364267 0 c -.03815 -9.05e-4 -.0745 .0075 -.106806 .02628 l -1.246395 .874095 c -.07112 .05846 -.07653 .19011 -.0092 .307341 6.2e-5 1.27e-4 1.56e-4 2.36e-4 2.49e-4 3.65e-4 9.4e-5 1.58e-4 1.56e-4 3.28e-4 2.49e-4 4.86e-4 .06804 .116859 .184711 .178028 .270908 .145977 l 1.38004 -.64254 c .129964 -.07489 .148612 -.288268 .03874 -.479228 -.08309 -.143407 -.213336 -.230056 -.327777 -.232771 z M 5.3120042 11.09811 c -.4244398 0 -.7613616 .119163 -.9450144 .22529 l .13442 .46019 c .1680816 -.105948 .4356595 -.193662 .6890692 -.193662 .4195399 -.0011 .4881692 .237283 .4881692 .390122 v .03613 c -.9141935 -.0014 -1.4916351 .314934 -1.4916351 .959794 0 .393697 .2939825 .762666 .8051205 .762666 .314655 0 .5777764 -.125535 .7354776 -.326622 h .015449 c 0 0 .1044518 .436706 .6798241 .269691 -.030228 -.181742 -.040022 -.375459 -.040022 -.608843 v -.898484 c 0 -.572632 -.2445259 -1.076271 -1.0708582 -1.076271 z m 4.0855133 0 c -.4273876 0 -.6194337 .216592 -.7359033 .40034 H 8.6514567 V 11.155101 H 7.9997329 v 2.526607 h .6865756 v -1.48172 c 0 -.06945 .00799 -.142898 .032236 -.2068 .056885 -.149189 .1956454 -.323703 .4172492 -.323703 .2770358 0 .4066656 .234212 .4066659 .572228 v 1.439995 h .6858464 v -1.499603 c 0 -.06636 .0091 -.14622 .02859 -.204853 .05639 -.169689 .206027 -.307767 .412018 -.307767 .280856 0 .415668 .230128 .415668 .627881 v 1.384342 h .686332 v -1.488046 c 0 -.784785 -.398454 -1.095552 -.848305 -1.095552 -.199223 0 -.356459 .04996 -.498631 .137217 -.119501 .07334 -.226545 .177533 -.320053 .314701 h -.0101 C 9.9852857 11.277751 9.7300686 11.09811 9.3975175 11.09811 Z m 3.7702045 0 c -.424357 0 -.761135 .119163 -.944954 .22529 l .134542 .46019 c .167916 -.105948 .435724 -.193662 .689009 -.193662 .419249 -.0011 .488047 .237283 .488047 .390122 v .03613 c -.914069 -.0014 -1.491635 .314935 -1.491635 .959794 0 .393697 .294179 .762666 .805607 .762666 .314406 0 .577581 -.125535 .734991 -.326622 h .01557 c 0 0 .104368 .436706 .679824 .269691 -.03027 -.181742 -.03996 -.375459 -.03996 -.608843 v -.898484 c 0 -.572632 -.244584 -1.076271 -1.071041 -1.076271 z m 2.851283 0 c -.262918 0 -.562046 .169705 -.687609 .533543 h -.0191 v -.476552 h -.618088 v 2.526607 h .704519 v -1.293715 c 0 -.06984 .0042 -.130692 .01527 -.186485 .05211 -.271032 .259455 -.444255 .556961 -.444255 .08167 0 .140095 .0088 .203333 .018 v -.661516 c -.05302 -.0107 -.0893 -.01563 -.155283 -.01563 z m 4.359463 1.220666 c -.01785 -2.49e-4 -.03513 .0026 -.05134 .0087 l -1.379902 .64181 c -.129425 .07518 -.148995 .288672 -.03838 .479836 .110283 .190599 .304693 .281175 .434159 .206435 l 1.246333 -.873486 c .07171 -.05929 .07696 -.191269 .0088 -.308619 l 6.07e-4 2.49e-4 c -.05507 -.09558 -.142925 -.153877 -.220278 -.154919 z m 2.079554 0 c -.07727 .001 -.164893 .05934 -.22012 .154917 l 4.86e-4 -2.49e-4 c -.06781 .11735 -.06254 .249333 .0088 .308619 l 1.246395 .873486 c .129217 .07474 .323803 -.01584 .434584 -.206436 .109867 -.191163 .09122 -.40466 -.03875 -.479836 l -1.380121 -.641809 c -.0162 -.0061 -.03344 -.0089 -.05128 -.0087 z M 5.7006055 12.463961 v .315188 c 0 .04661 -.00406 .09467 -.01697 .136731 -.052858 .175062 -.2341485 .323033 -.4608596 .323033 -.1890087 0 -.3391518 -.10731 -.3391518 -.334225 0 -.347067 .3820367 -.442997 .8169814 -.440727 z m 7.8557785 0 v .315188 c 0 .04661 -.0041 .09467 -.01691 .136731 -.05286 .175062 -.234209 .323033 -.46092 .323033 -.189051 0 -.339212 -.10731 -.339213 -.334225 0 -.347067 .382097 -.442997 .817043 -.440727 z m 7.861374 .483243 c -.135058 1.56e-4 -.246716 .07024 -.26148 .161365 l -.133933 1.516207 c 0 .150145 .175357 .273037 .395839 .273037 .220898 0 .396448 -.122892 .396448 -.273037 l -.134238 -1.516207 c -.01509 -.09113 -.126594 -.161196 -.262149 -.161365 z" />
  <path id="source" fill="#fff" d="M21.418184 9.1021869 c -.220482 0 -.395839 .1230808 -.395839 .2729758 l .133933 1.5162683 c.01478 .09132 .126688 .161852 .261967 .161852 .135555 -1.72e-4 .247054 -.07065 .262149 -.161852 l .134238 -1.5162683 c 0 -.149895 -.17555 -.2729758 -.396448 -.2729758 z M -1.4486275e-5 10.318413 c 0 0 .597125316275 2.446983 .692414506275 2.844167  .11123373 .463624 .31170438 .63419 .88954338 .519128 l .3729088 -1.51718 c .094496 -.376831 .157901 -.64553 .2186604 -1.02883 h .010644 c .042693 .387111 .1035396 .653142 .1813148 1.030168 0 0 .1517616 .688693 .229487 1.050421 .07795 .361633 .2948212 .589502 .8608346 .465421 L 4.3443025 10.318413 H 3.6269506 L 3.323563 11.77252 c -.081554 .423252 -.1554409 .75396 -.2125173 1.14117 H 3.1008274 C 3.0489413 12.530225 2.9830587 12.21273 2.9001099 11.800377 L 2.5842536 10.318413 H 1.8368546 L 1.499041 11.763032 c -.09568 .438984 -.1852681 .793372 -.2421991 1.167445 h -.010218 c -.058322 -.35224 -.1359895 -.797691 -.2200594 -1.222612 0 0 -.20074192 -1.033275 -.27115107 -1.389452 z m 6.838444886275 0 v 3.363295 h .6829868 v -.714067 -2.649228 z m 9.6188076 0 v 2.481173 c 0 .342022 .06439 .581706 .202117 .728117 .120331 .128055 .318542 .210936 .556048 .210936 .201923 0 .400619 -.0383 .494252 -.07317 l -.0088 -.533665 c -.06968 .01711 -.149794 .03078 -.259413 .03078 -.232648 0 -.310686 -.148997 -.310686 -.456115 v -.949212 h .594793 v -.643512 h -.594812 v -.795328 z m 2.778843 0 c -.114465 .0027 -.244518 .08936 -.327231 .232772 -.110615 .19096 -.09105 .404339 .03838 .479228 l 1.379902 .642539 c .0862 .03205 .203231 -.02912 .27103 -.145976 .06814 -.117446 .06289 -.249593 -.0088 -.308193 l -1.246353 -.874094 c -.03237 -.01875 -.06877 -.02718 -.106928 -.02628 z m 4.364267 0 c -.03815 -9.05e-4 -.0745 .0075 -.106806 .02628 l -1.246395 .874095 c -.07112 .05846 -.07653 .19011 -.0092 .307341 6.2e-5 1.27e-4 1.56e-4 2.36e-4 2.49e-4 3.65e-4 9.4e-5 1.58e-4 1.56e-4 3.28e-4 2.49e-4 4.86e-4 .06804 .116859 .184711 .178028 .270908 .145977 l 1.38004 -.64254 c .129964 -.07489 .148612 -.288268 .03874 -.479228 -.08309 -.143407 -.213336 -.230056 -.327777 -.232771 z M 5.3120042 11.09811 c -.4244398 0 -.7613616 .119163 -.9450144 .22529 l .13442 .46019 c .1680816 -.105948 .4356595 -.193662 .6890692 -.193662 .4195399 -.0011 .4881692 .237283 .4881692 .390122 v .03613 c -.9141935 -.0014 -1.4916351 .314934 -1.4916351 .959794 0 .393697 .2939825 .762666 .8051205 .762666 .314655 0 .5777764 -.125535 .7354776 -.326622 h .015449 c 0 0 .1044518 .436706 .6798241 .269691 -.030228 -.181742 -.040022 -.375459 -.040022 -.608843 v -.898484 c 0 -.572632 -.2445259 -1.076271 -1.0708582 -1.076271 z m 4.0855133 0 c -.4273876 0 -.6194337 .216592 -.7359033 .40034 H 8.6514567 V 11.155101 H 7.9997329 v 2.526607 h .6865756 v -1.48172 c 0 -.06945 .00799 -.142898 .032236 -.2068 .056885 -.149189 .1956454 -.323703 .4172492 -.323703 .2770358 0 .4066656 .234212 .4066659 .572228 v 1.439995 h .6858464 v -1.499603 c 0 -.06636 .0091 -.14622 .02859 -.204853 .05639 -.169689 .206027 -.307767 .412018 -.307767 .280856 0 .415668 .230128 .415668 .627881 v 1.384342 h .686332 v -1.488046 c 0 -.784785 -.398454 -1.095552 -.848305 -1.095552 -.199223 0 -.356459 .04996 -.498631 .137217 -.119501 .07334 -.226545 .177533 -.320053 .314701 h -.0101 C 9.9852857 11.277751 9.7300686 11.09811 9.3975175 11.09811 Z m 3.7702045 0 c -.424357 0 -.761135 .119163 -.944954 .22529 l .134542 .46019 c .167916 -.105948 .435724 -.193662 .689009 -.193662 .419249 -.0011 .488047 .237283 .488047 .390122 v .03613 c -.914069 -.0014 -1.491635 .314935 -1.491635 .959794 0 .393697 .294179 .762666 .805607 .762666 .314406 0 .577581 -.125535 .734991 -.326622 h .01557 c 0 0 .104368 .436706 .679824 .269691 -.03027 -.181742 -.03996 -.375459 -.03996 -.608843 v -.898484 c 0 -.572632 -.244584 -1.076271 -1.071041 -1.076271 z m 2.851283 0 c -.262918 0 -.562046 .169705 -.687609 .533543 h -.0191 v -.476552 h -.618088 v 2.526607 h .704519 v -1.293715 c 0 -.06984 .0042 -.130692 .01527 -.186485 .05211 -.271032 .259455 -.444255 .556961 -.444255 .08167 0 .140095 .0088 .203333 .018 v -.661516 c -.05302 -.0107 -.0893 -.01563 -.155283 -.01563 z m 4.359463 1.220666 c -.01785 -2.49e-4 -.03513 .0026 -.05134 .0087 l -1.379902 .64181 c -.129425 .07518 -.148995 .288672 -.03838 .479836 .110283 .190599 .304693 .281175 .434159 .206435 l 1.246333 -.873486 c .07171 -.05929 .07696 -.191269 .0088 -.308619 l 6.07e-4 2.49e-4 c -.05507 -.09558 -.142925 -.153877 -.220278 -.154919 z m 2.079554 0 c -.07727 .001 -.164893 .05934 -.22012 .154917 l 4.86e-4 -2.49e-4 c -.06781 .11735 -.06254 .249333 .0088 .308619 l 1.246395 .873486 c .129217 .07474 .323803 -.01584 .434584 -.206436 .109867 -.191163 .09122 -.40466 -.03875 -.479836 l -1.380121 -.641809 c -.0162 -.0061 -.03344 -.0089 -.05128 -.0087 z M 5.7006055 12.463961 v .315188 c 0 .04661 -.00406 .09467 -.01697 .136731 -.052858 .175062 -.2341485 .323033 -.4608596 .323033 -.1890087 0 -.3391518 -.10731 -.3391518 -.334225 0 -.347067 .3820367 -.442997 .8169814 -.440727 z m 7.8557785 0 v .315188 c 0 .04661 -.0041 .09467 -.01691 .136731 -.05286 .175062 -.234209 .323033 -.46092 .323033 -.189051 0 -.339212 -.10731 -.339213 -.334225 0 -.347067 .382097 -.442997 .817043 -.440727 z m 7.861374 .483243 c -.135058 1.56e-4 -.246716 .07024 -.26148 .161365 l -.133933 1.516207 c 0 .150145 .175357 .273037 .395839 .273037 .220898 0 .396448 -.122892 .396448 -.273037 l -.134238 -1.516207 c -.01509 -.09113 -.126594 -.161196 -.262149 -.161365 z" />
</svg>`


Notice that both paths are **exactly the same**. So based on this, I think this result is good enough for merging. It's the same exact path and everything, the fact that inkscape prints an outline I don't understand, maybe it's a setting I don't know or something .But this is as 100% similar as it gets.

jorgeamadosoria replied:
This is the optimized svg code that I compared with the source and produced just the outline as seen above:
`<svg role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title>Walmart</title><path d="M21.41818 9.10219c-.22048 0-.39583.12308-.39583.27297l.13393 1.51627c.01478.09132.12669.16185.26197.16185.13555-.00017.24705-.07065.26214-.16185l.13424-1.51627c0-.1499-.17555-.27297-.39645-.27297zM-.00002 10.3184s.59713 2.44699.69242 2.84417c.11123.46362.3117.63419.88954.51913l.37291-1.51718c.0945-.37683.1579-.64553.21866-1.02883h.01065c.04269.3871.10354.65314.18131 1.03017 0 0 .15176.68869.22949 1.05042.07795.36163.29482.5895.86083.46542l.88851-3.3633h-.71735l-.30339 1.45411c-.08155.42325-.15544.75396-.21251 1.14117h-.01022c-.05189-.38347-.11777-.70096-.20072-1.11331l-.31586-1.48197h-.7474l-.3378 1.44462c-.09569.43899-.18528.79337-.2422 1.16745h-.01023c-.05832-.35224-.13599-.7977-.22006-1.22261 0 0-.20074-1.03328-.27115-1.38946zm6.83845 0v3.3633h.68299v-3.3633zm9.6188 0v2.48118c0 .34202.0644.5817.20213.72811.12033.12806.31854.21094.55604.21094.20193 0 .40062-.0383.49426-.07317l-.0088-.53367c-.06968.01711-.1498.03078-.25942.03078-.23265 0-.31068-.149-.31068-.45611v-.94921h.59479v-.64351h-.59481v-.79533zm2.77885 0c-.11446.0027-.24452.08936-.32723.23277-.11062.19096-.09105.40434.03838.47923l1.3799.64254c.0862.03205.20323-.02912.27103-.14597.06814-.11745.0629-.2496-.0088-.3082l-1.24635-.8741c-.03237-.01874-.06877-.02717-.10693-.02627zm4.36427 0c-.03815-.0009-.0745.0075-.1068.02628l-1.2464.8741c-.07112.05846-.07653.1901-.0092.30734.00006.00013.00015.00023.00025.00036.00009.00016.00015.00033.00024.00049.06804.11686.18472.17803.27091.14598l1.38004-.64254c.12997-.0749.14861-.28827.03874-.47923-.08309-.1434-.21333-.23006-.32777-.23277zM5.312 11.0981c-.42444 0-.76136.11916-.94501.22529l.13442.46019c.16808-.10595.43566-.19366.68907-.19366.41954-.0011.48817.23728.48817.39012v.03613c-.9142-.0014-1.49164.31493-1.49164.9598 0 .3937.29399.76266.80512.76266.31466 0 .57778-.12554.73548-.32662h.01545s.10445.4367.67982.26969c-.03022-.18174-.04002-.37546-.04002-.60884v-.89849c0-.57263-.24452-1.07627-1.07086-1.07627zm4.08552 0c-.42739 0-.61944.2166-.7359.40034h-.01016v-.34335h-.65173v2.5266h.68658V12.2c0-.06945.00799-.1429.03223-.2068.05689-.1492.19565-.3237.41725-.3237.27704 0 .40667.2342.40667.57222v1.44h.68585v-1.4996c0-.06636.0091-.14622.02859-.20486.05639-.16969.20602-.30776.41201-.30776.28086 0 .41567.23012.41567.62788v1.38434h.68633v-1.48805c0-.78478-.39845-1.09555-.8483-1.09555-.19922 0-.35646.04996-.49863.13722-.1195.07334-.22655.17753-.32006.3147h-.0101c-.10853-.27228-.36375-.45192-.6963-.45192zm3.7702 0c-.42435 0-.76113.11916-.94495.22529l.13454.46019c.16792-.10595.43572-.19366.689-.19366.41926-.0011.48806.23728.48806.39012v.03613c-.91407-.0014-1.49164.31494-1.49164.9598 0 .3937.29418.76266.8056.76266.31441 0 .57759-.12554.735-.32662h.01557s.10437.4367.67982.26969c-.03027-.18174-.03996-.37546-.03996-.60884v-.89849c0-.57263-.24458-1.07627-1.07104-1.07627zm2.85129 0c-.26292 0-.56205.1697-.68761.53354h-.0191v-.47655h-.6181v2.5266h.70453V12.388c0-.06985.0042-.1307.01527-.1865.0521-.27102.25945-.44425.55696-.44425.08167 0 .1401.0088.20333.018v-.66151c-.05302-.0107-.0893-.01563-.15528-.01563zm4.35946 1.22067c-.01785-.00025-.03513.0026-.05134.0087l-1.3799.6418c-.12943.07519-.149.28868-.03838.47984.11028.1906.30469.28118.43415.20644l1.24634-.87349c.0717-.05929.07696-.19127.0088-.30862l.0006.00025c-.05507-.09558-.14292-.15388-.22027-.15492zm2.07955 0c-.07727.001-.1649.05934-.22012.15491l.00049-.00025c-.06781.11735-.06254.24934.0088.30862l1.2464.87349c.12921.07474.3238-.01584.43458-.20644.10986-.19116.09122-.40466-.03875-.47983l-1.38012-.64181c-.0162-.0061-.03344-.0089-.05128-.0087zm-16.75741.14518v.31519c0 .0466-.00406.09467-.01697.13673-.05286.17506-.23415.32303-.46086.32303-.18901 0-.33916-.1073-.33916-.33422 0-.34707.38204-.443.81699-.44073zm7.85577 0v.31519c0 .0466-.0041.09467-.0169.13673-.05287.17506-.23421.32303-.46093.32303-.18905 0-.3392-.1073-.3392-.33422 0-.34707.38209-.443.81703-.44073zm7.86138.48324c-.13506.00016-.24672.07024-.26148.16137l-.13393 1.5162c0 .15015.17535.27304.39583.27304.2209 0 .39645-.12289.39645-.27303l-.13424-1.51621c-.01509-.09113-.12659-.1612-.26214-.16137z"/></svg>`

I think this is good enough. it's possible that a better fit may exist, but this is a very good fit and as seen before, even with the same exact paths the outline still shows up for me.
I vote to merge this one.

ericcornelissen replied:
The goal of the comparison is not to see if you get the same out as the contributor, it's to compare it to the original. That is, we want to compare the SVG being contributed to the unoptimized (only rescaled) original SVG. Of course, in this case, the `(R)` has to be removed before comparing.

If I do that, I get [the comparison I first shared](https://github.com/simple-icons/simple-icons/pull/6269#discussion_r690987599). And if I understand you correctly, you redid the optimization before comparing. Is that correct @jorgeamadosoria ?

jorgeamadosoria replied:
notice that the comparison I posted where the paths are the same but with different colors is the **unoptimized source**, compare to itself.

ericcornelissen replied:
You compared the SVG to itself? :confused: 

Do you mean to say you compared the unoptimized source to the SVG in the Pull Request? Where "unopitmized source" means [this SVG](https://cdn.corporate.walmart.com/12/9d/06ecf3584e91acccc37162f09714/logowhite.svg) but resized and with the `(R)` removed?

jorgeamadosoria replied:
I compared the source svg to itself, and it gave me the same diff that the comparison between source and optimized svg gave.

Meaning that the optimized svg is as close as possible to the source as it could be.

And yes, unoptimized source is the logowhite resized with the R removed.

ericcornelissen replied:
Still a bit confused about your process. 

> I compared the source svg to itself

So you compared [`logowhite.svg`](https://cdn.corporate.walmart.com/12/9d/06ecf3584e91acccc37162f09714/logowhite.svg) to [`logowhite.svg`](https://cdn.corporate.walmart.com/12/9d/06ecf3584e91acccc37162f09714/logowhite.svg). That should indeed line up perfectly as it's the same file, but I don't understand why this would be useful.


> it gave me the same diff that the comparison between source and optimized svg gave.

So you also compared [`logowhite.svg`](https://cdn.corporate.walmart.com/12/9d/06ecf3584e91acccc37162f09714/logowhite.svg) (with the `(R)` removed) to [`walmart.svg`](https://github.com/sachinraja/simple-icons/blob/a28992f379f4286fda3eafe1761e6a7ad3af625c/icons/walmart.svg) (i.e. the SVG currently being added in this Pull Request)? If so, that is indeed the comparison we're interested in, is this one of the screenshot you shared? If so, could you link to that exact comment or reshare the screenshot?

jorgeamadosoria replied:
OK, this is why I did these comparisons:
- logo white vs logo white to see what it looks like when the match is perfect. It could be the case that you could either not see anything (since they are perfectly aligned and color-inverted) or you could see the outline. As it turns out, inkscape shows the outline of two icons even if they are perfectly aligned.
- Armed with that knowledge, I made the second comparison. The result is the same as the first one. Thus, that means that walmart.svg is as close as it can be to logowhite.svg, and the outline shown is inkscape's way of saying that there is a 100% match between them.

That is, of course, if none of the outline marks is thicker than the others.

ericcornelissen replied:
Okay, I understand your approach now. I'm pretty sure I did the exact same thing as you did for your second bullet to produce the comparison in [the comment that started this thread](https://github.com/simple-icons/simple-icons/pull/6269#discussion_r690987599). I can try to do another comparison later today.

However, you [did point out you say the same/similar differences](https://github.com/simple-icons/simple-icons/pull/6269#discussion_r691752050), what happened with that?

jorgeamadosoria replied:
A mistake, for sure! There is no reason for that.

ericcornelissen replied:
@jorgeamadosoria: did you compare `logowhite.svg` to the SVG that is currently (i.e. in [this commit](https://github.com/sachinraja/simple-icons/tree/a28992f379f4286fda3eafe1761e6a7ad3af625c) being added in this Pull Request, yes or no?

jorgeamadosoria replied:
I have lost the thread of this thread, so to speak. It's too confusing now.

To summarize:
The SVG in https://github.com/simple-icons/simple-icons/pull/6269#discussion_r693434268 is the optimized SVG that I think is good enough to merge.

ericcornelissen replied:
Alright, seems to confusion came from the fact that you we're talking about an SVG that you included in a comment while I was still wondering about the SVG in the Pull Request.

Comparing the SVG in [the comment you mentioned](https://github.com/simple-icons/simple-icons/pull/6269#discussion_r693434268) to `logowhite.svg` looks very good, nice work :+1: @sachinraja could you update the Pull Request with that SVG?

sachinraja replied:
Updated! Is this ready to be merged?

ericcornelissen commented on _data/simple-icons.json:
```diff
@@ -9956,6 +9956,12 @@
+            "title": "Walmart",
+            "hex": "0071CE",
+            "source": "https://cdn.corporate.walmart.com/12/9d/06ecf3584e91acccc37162f09714/logowhite.svg",
+            "guidelines": "https://one.walmart.com/content/people-experience/associate-brand-center.html.html"
```
Not sure where the double `.html` comes from but it seems to work with one as well

```suggestion
            "guidelines": "https://one.walmart.com/content/people-experience/associate-brand-center.html"
```

sachinraja commented:
![walmart](https://user-images.githubusercontent.com/58836760/131258411-e6cc111c-7e94-4bf7-b8a5-e1103e961783.png)
Here's the preview for latest version of the icon

ericcornelissen reviewed (commented):
The SVG and metadata look good now, can someone remind me why the trademark symbol is not included?

sachinraja commented:
@ericcornelissen Walmart does not seem to require it for out-of-context logos in their [guidelines](https://one.walmart.com/content/people-experience/associate-brand-center.html) (in the "Logo creation" section).

ericcornelissen reviewed (approved):
In that case, this looks good to me :+1: If it turns out we should include the trademark symbol after all we can update it later

sachinraja commented:
Thank you @jorgeamadosoria for your work on this!

dirien commented:
Soo cool @sachinraja! Love it!
